src/filters/simple-filter/filter-fields-component.js

```diff
@@ -88,6 +88,8 @@ angular.module('patternfly.filters').component('pfFilterFields', {
       if (keyEvent.which === 13) {
         ctrl.addFilterFn(ctrl.currentField, ctrl.currentValue);
         ctrl.currentValue = undefined;
+        keyEvent.stopPropagation();
+        keyEvent.preventDefault();
```

Comment:
This fixes the issue when hitting [Enter] on an input filter switches to first/List view from whatever view you were on.